### PR TITLE
feat: [PATRON2020-6] add dashboard layout view

### PR DESCRIPTION
**Pull request by MartaJaszewska**

### CODE REVIEW

File: frontend/src/views/Dashboard.jsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+const Dashboard = () => {
+  const classes = useStyles()
+  return (
+    <Grid container maxwidth='xs' className={classes.root}>
+      <Grid item xs={12} className={classes.navigation}>
+        NAVIGATION
+      </Grid>
+      <Grid container maxwidth='xs' className={classes.content}>
+        <Grid
```
Comment (jwieckowski) [inline frontend/src/views/Dashboard.jsx]: I think we should move home map and sensors list to another component, because of the fact that all sites will contain navigation header and only content below will change. If you change code here to display `props.children`, it might be more versatile solution

File: frontend/src/views/Dashboard.jsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    boxSizing: 'border-box'
+  },
+  map: {
+    backgroundColor: 'white'
+  },
+  image: {
+    maxWidth: '100%',
+    height: 'calc(100vh - 30px - 100px)', // change navigation height if needed
+    objectFit: 'contain',
+    margin: '15px 0'
```
Comment (kamiljalowiczor) [inline frontend/src/views/Dashboard.jsx]: Margin here is unnecessary and it causes the page to be scrollable. If this margin is really needed for something, you could try using overflow: hidden on root class instead (although it may produce an undesired side effect on sensors list component later on).

Comment (MartaJaszewska) [reply frontend/src/views/Dashboard.jsx]: Thanks, I didn't notice it. I'm going to delete this margin.

File: frontend/src/components/DashboardContent/DashboardContent.jsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import SensorsList from '../SensorsList'
+
+const useStyles = makeStyles(theme => ({
+  content: {
+    height: 'calc(100vh - 150px)', // check navigation height in frontend\src\components\Navigation\Navigation.jsx if needed
+    boxSizing: 'border-box'
+  }
+}))
+
+const DashboardContent = ({ children }) => {
```
Comment (hgaluszka) [inline frontend/src/components/DashboardContent/DashboardContent.jsx]: Children are unused.

File: frontend/src/components/SmartHomeMap/SmartHomeMap.jsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+      container
+      justify='center'
+      alignItems='center'
+    >
+      <div className={classes.mapContainer}> {/* use or replace */}
+        {/* Map Component */}
+        {/* delete img tag below */}
+        <img
+          src={map}
+          alt='Your Smart Home Map'
```
Comment (kamiljalowiczor) [inline frontend/src/components/SmartHomeMap/SmartHomeMap.jsx]: The mock image you are using is 800x801px, which (on my screen) is more than intended height of mapContainer. You should make sure that the image always fits within its container, and also make sure that the page is not scrollable.

File: frontend/src/components/DashboardContent/DashboardContent.jsx [hunk 5]
```diff
@@ -0,0 +1,23 @@
+import React from 'react'
+import { Grid, makeStyles } from '@material-ui/core'
+import SmartHomeMap from '../SmartHomeMap'
+import SensorsList from '../SensorsList'
+
+const useStyles = makeStyles(theme => ({
```
Comment (hgaluszka) [inline frontend/src/components/DashboardContent/DashboardContent.jsx]: Theme is unused.

File: frontend/src/components/SmartHomeMap/SmartHomeMap.jsx [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+import { Grid, makeStyles } from '@material-ui/core'
+import map from '../../assets/changeMe.jpg' // delete after changing img tag below
+
+const useStyles = makeStyles(theme => ({
+  mapBackground: {
+    backgroundColor: 'white'
+  },
+  mapContainer: { // use or replace
+    maxWidth: '100%',
+    height: 'calc(100vh - 150px)', // change navigation height if needed
```
Comment (hgaluszka) [inline frontend/src/components/SmartHomeMap/SmartHomeMap.jsx]: I my opinion this calculation should be defined as const.

File: frontend/src/components/SmartHomeMap/SmartHomeMap.jsx (hunk 4 above)
Comment (MartaJaszewska) [reply frontend/src/components/SmartHomeMap/SmartHomeMap.jsx]: I think it doesn't matter yet, because there will be a map component with its own styling, not an img tag.

File: frontend/src/components/SmartHomeMap/SmartHomeMap.jsx (hunk 6 above)
Comment (MartaJaszewska) [reply frontend/src/components/SmartHomeMap/SmartHomeMap.jsx]: I don't think it matters when I use it in one place only.

File: frontend/src/views/Dashboard.jsx [hunk 7]
```diff
@@ -1,21 +1,25 @@
...
-    <Grid container maxwidth='xs' className={classes.root}>
+    <Grid container maxwidth='xs' className={classes.root} data-testid='dashboard-id'>
       <Navigation />
-      <DashboardContent />
+      <Grid container maxwidth='xs' className={classes.content}>
+        {children}
+      </Grid>
     </Grid>
   )
 }
```
Comment (kamiljalowiczor) [inline frontend/src/views/Dashboard.jsx]: I just realized that it would be way more fitting if this component was named Layout. It's gonna be used for every page and not just dashboard. Renaming it now may save us from some confusion in the future.

File: tests-server/authors/AuthorMarta.spec.js [hunk 8]
```diff
@@ -0,0 +1,38 @@
+/* globals beforeAll, afterAll, describe, test, expect */
```
Comment (makeros) [inline tests-server/authors/AuthorMarta.spec.js]: This should be at least in a separate commit ;)
